mlflow/pytorch/torchserve/deploy/serve.py

```diff
@@ -0,0 +1,264 @@
+            with open(config_properties, "r") as f:
+                lines = f.readlines()
+                for line in lines:
+                    name = line.strip().split("=")
+                    if name[0] == address_strings[0] and name[1] is not None:
+                        inference_port = name[1]
+                    if name[0] == address_strings[1] and name[1] is not None:
+                        management_port = name[1]
+        return inference_port, management_port
+
+    def validate_mandatory_arguments(self):
+        """
+        Validate the mandatory arguments is present if not raise exception
+        """
+
+        if not self.server_config["version"]:
+            raise Exception("Environment Variable VERSION - missing")
+
+        if not self.server_config["model_file"]:
+            raise Exception("Environment Variable MODEL_FILE - missing")
+
+        if not self.server_config["handler_file"]:
+            raise Exception("Environment Variable HANDLER_FILE - missing")
+
+    def create_deployment(self, name, model_uri, flavor=None, config=None):
+        """
+        Deploy the model at the model_uri to the specified target
+        """
+        self.validate_mandatory_arguments()
+        mar_file_path = self.generate_mar_file(
+            model_name=name,
+            version=self.server_config["version"],
+            model_file=self.server_config["model_file"],
+            handler_file=self.server_config["handler_file"],
+            extra_files=self.server_config["extra_files"],
+            model_uri=model_uri,
+        )
+
+        self.register_model(
+            mar_file_path=mar_file_path,
+            name=name,
+            model_uri=model_uri,
+            flavor=flavor,
+            config=config,
+        )
+        return {"name": name, "flavor": flavor}
+
+    def delete_deployment(self, name):
+        """
+        Delete the deployment with the name given at --name from the specified target
+        """
+
+        version = self.server_config["version"]
+        url = "{}/{}/{}/{}".format(self.management_api, "models", name, version)
+        resp = requests.delete(url)
+        if resp.status_code != 200:
+            raise Exception("Unable to list deployments")
+        return None
+
+    def update_deployment(self, name, model_uri=None, flavor=None, config=None):
+        """
+        Update the deployment with the name given at --name from the specified target
+
+        Using -C or --config additional parameters shall be updated for the corresponding model
+        """
+
+        query_path = ""
+
+        if config is not None:
+            for key in config:
+                query_path += "&" + key + "=" + str(config[key])
+
+            query_path = query_path[1:]
+
+        url = "{}/{}/{}?{}".format(self.management_api, "models", name, query_path)
+        resp = requests.put(url)
+
+        if resp.status_code != 202:
+            raise Exception("Unable to list deployments")
```

Comment:
```suggestion
            raise Exception("Unable to update deployment with name %s. Server returned status code %s and response: %s" % (name, resp.status_code, resp.content))
```

Maybe something like this to make failures easier to debug?